mantidproject/mantid
# Computes phonon density of states in the incoherent approximation.

This is a python algorithm to compute the phonon density of states from a powder S(q,w) dataset. In this case, all the scattering can be treated as incoherent (the coherent scattering averaged over all angles in a powder is similar to the isotropic incoherent scattering) and a simple formula can be applied to derive the neutron weighted partial density of states. The theory is partly detailed in the documentation rst.

It only operates on histogram workspaces with one axis being momentum transfer (Q) and one being energy transfer (in either meV or cm). By default it outputs the DOS g(E) in meV but can be specified to output in wavenumbers (cm^-1).

**To test:**

By constructing a (Q,w) workspace

```
ws = CreateSampleWorkspace(binWidth = 0.1, XMin = 0, XMax = 50, XUnit = 'DeltaE')
ws = ScaleX(ws, -25, "Add")
LoadInstrument(ws, InstrumentName='MARI', RewriteSpectraMap = True)
ws = SofQW(ws, [0, 0.05, 8], 'Direct', 25)
ws_DOS = ComputeIncoherentDOS(ws)
```

Alternatively, with a datafile (note the datafiles should be accessible in the archive, but the mapfile might need to be copied somewhere accessible. It is here: https://svn.isis.rl.ac.uk/InstrumentFiles/trunk/mari/mari_res2013.map 
The following uses a measurement of Aluminium on MARI at ISIS.

```
from Direct import DirectEnergyConversion
rd = DirectEnergyConversion.DirectEnergyConversion('MARI')
ws = rd.convert_to_energy(21334, 21335, 60, [-55,0.05,55], 'mari_res2013.map', 
    monovan_run=21347, sample_mass=106.4, sample_rmm=27, monovan_mapfile='mari_res2013.map')
ws_sqw = SofQW3(ws, [0,0.1,12], 'Direct', 60)
SetSampleMaterial(ws_sqw,'Al')
ws_dos = ComputeIncoherentDOS(ws_sqw, Temperature=5, StatesPerEnergy=True)

```

Compare to the data here: http://journals.aps.org/prb/abstract/10.1103/PhysRevB.77.024301
Too high by a factor of around 3, which I haven't figured out why... 
Possibly from the Debye-Waller factor. Or possibly due to multiple scattering.



Fixes #15843.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
